### PR TITLE
coll/han: fix bug in reduce in place

### PR DESCRIPTION
**Pull request by wzamazon**
Piror to this patch, the reduce code try to applied arithematic operations on a sendbuf, even when it
is MPI_IN_PLACE. this patch addressed the issue.

### CODE REVIEW

**Comment by wzamazon** [comment]
This appears to be a bug in han's reduce, which does to handle MPI_IN_PLACE correctly.

https://github.com/open-mpi/ompi/pull/11800 should fix it

**Review by wenduwan — APPROVED** [review]
Thank you!

**Comment by jsquyres** [comment]
@B-a-S Looks like a failure in the NVIDIA CI environment:

```
Error from server (NotFound): pods "ompi-ci-1" not found
```

Can you please investigate?

File: ompi/mca/coll/han/coll_han_reduce.c [hunk 1]
```diff
@@ -172,7 +172,8 @@ mca_coll_han_reduce_intra(const void *sbuf,
         mca_coll_task_t *t_next_seg = OBJ_NEW(mca_coll_task_t);
         /* Setup up t_next_seg task arguments */
         t->cur_task = t_next_seg;
-        t->sbuf = (char *) t->sbuf + extent * t->seg_count;
+        if (t->sbuf != MPI_IN_PLACE)
```
Comment (bwbarrett) [inline ompi/mca/coll/han/coll_han_reduce.c:175]: We use curly braces around if bodies (even single line bodies), like all rational thinking people.

**Review by bwbarrett — CHANGES_REQUESTED** [review]

File: ompi/mca/coll/han/coll_han_reduce.c (hunk 1 above)
Comment (wzamazon) [reply ompi/mca/coll/han/coll_han_reduce.c]: Thank you! Fixed in new revision.

**Comment by jsquyres** [comment]
CI infra issue has been resolved.

@bosilca Can you review?

**Review by bwbarrett — APPROVED** [review]

**Comment by wzamazon** [comment]
all CI passed. Ready for merge!

**Comment by jsquyres** [comment]
@wzamazon It says you force-pushed an hour ago.  What did you push after the PR was approved?

**Comment by wzamazon** [comment]
> @wzamazon It says you force-pushed an hour ago. What did you push after the PR was approved?

I rebased to latest main branch (using the github's rebase branch option)

**Pull request merged**